core/src/main/java/ar/com/wolox/wolmo/core/util/ImageProvider.kt

```diff
@@ -0,0 +1,247 @@
+import androidx.fragment.app.Fragment
+import ar.com.wolox.wolmo.core.di.scopes.ApplicationScope
+import java.io.ByteArrayOutputStream
+import java.io.File
+import java.io.IOException
+import javax.inject.Inject
+
+/**
+ * Utils class to manipulate images, through {@link Bitmap}s or their corresponding {@link Uri}, and
+ * for retrieving pictures from gallery/taking them from the camera.
+ */
+@ApplicationScope
+class ImageProvider @Inject constructor(val context: Context, val toastFactory: ToastFactory, val wolmoFileProvider: WolmoFileProvider) {
+
+    @Retention(AnnotationRetention.SOURCE)
+    @StringDef(PNG, JPG)
+    annotation class ImageFormat
+
+    /*
+     * Image compression formats supported.
+     */
+
+    companion object {
+        const val PNG = "png"
+        const val JPG = "jpg"
+
+        /**
+         * Prevents quality from being outside 0...100 range.
+         *
+         * @param quality target quality
+         * @return if below 0, returns 0, if above 100, return 100, else returns `quality` param
+         */
+        private fun sanitizeQuality(quality: Int): Int = when {
+            quality < 0 -> 0
+            quality > 100 -> 100
+            else -> quality
+        }
+
+        /**
+         * Re-sizes the image, represented as a [Bitmap], to fit the boundaries and keeping its
+         * aspect ratio.
+         *
+         * @param image     [Bitmap] to scale
+         * @param maxWidth  max width it can occupy
+         * @param maxHeight max height it can occupy
+         * @return If the re-scaling was necessary, the scaled [Bitmap]. Else, it returns the
+         * target one.
+         */
+        fun fit(image: Bitmap, maxWidth: Int, maxHeight: Int): Bitmap {
+            val width = image.width
+            val height = image.height
+
+            if (maxWidth <= 0 || maxHeight <= 0 || width <= maxWidth && height <= maxHeight) {
+                return image
+            }
+
+            val ratioImage = width.toFloat() / height.toFloat()
+
+            var finalWidth = maxWidth
+            var finalHeight = maxHeight
+            if (ratioImage > 1) {
+                finalHeight = (finalWidth.toFloat() / ratioImage).toInt()
+            } else {
+                finalWidth = (finalHeight.toFloat() * ratioImage).toInt()
+            }
+
+            return Bitmap.createScaledBitmap(image, finalWidth, finalHeight, true)
+        }
+
+        /**
+         * Get [byte[]] from an image [File]
+         *
+         * @param file      target image file
+         * @param format    image compress format
+         * @param quality   compress quality, between 0 and 100
+         * @param maxWidth  max width of the target image
+         * @param maxHeight max height of the target image
+         * @return byte array with the formatted information of the image file, if the image exceeded
+         * boundaries, it's re scaled.
+         */
+        fun getImageAsByteArray(
```

Comment:
You could avoid return type and make it a one line function.

Reply:
ok :+1: